HelpSource/Classes/Function.schelp

```diff
@@ -122,10 +122,65 @@ e = Environment.make({ ~a = 3; ~b = 10 });
 )
 ::
 
+method::valueWith
```

Comment:
I don't particularly like the names, partcularly since doesNotUnderstandWithKeys no longer exists and performAbout sounds odd... but I can't think of a better one!

Reply:
well, I wouldn't see this method too closely connected to `doesNotUnderstand`, but read it in connection with the other similar methods. The naming is not very systematic:

```superocllider
performList { arg selector, arglist;
performWithEnvir { |selector, envir|
performKeyValuePairs { |selector, pairs|
multiChannelPerform { arg selector ... args;
```

Then, we can ask: is the method one that will be used mainly in special points in the class library, or will it be a general purpose thing? I think the former, because usually you'll have either an array of values or keywords, right?

Very systematic would be: `performListAndKeyValuePairs`. But you'd want to use this only internally really.

Another possibility is to turn round the argument order and extend (and reimplement) `performKeyValuePairs`:
```supercollider
performKeyValuePairs { |selector, pairs, positionalArgs|
```

Reply:
(but the latter would involve more testing, because it changes the way this method works)

Reply:
I agree, done!